launch/launcher.go

```diff
@@ -124,6 +146,85 @@ func (l *Launcher) env(process Process) error {
 	})
 }
 
+func (l *Launcher) execD(process Process) error {
+	execDs, err := l.buildpackFiles(process, ExecDDirName, supportsExecD)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find all exec.d executables in layers dir, '%s'", l.LayersDir)
+	}
+	for _, execD := range execDs {
+		if err := l.ExecD.ExecD(execD, l.Env); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func supportsExecD(bp Buildpack) bool {
+	if bp.API == "" {
+		return false
+	}
+	return api.MustParse(bp.API).Compare(api.MustParse("0.5")) >= 0
+}
+
+func (l *Launcher) buildpackFiles(process Process, dirName string, filters ...func(Buildpack) bool) ([]string, error) {
+	var files []string
+
+	appendIfFile := func(path string, fi os.FileInfo) {
+		if !fi.IsDir() {
+			files = append(files, path)
+		}
+	}
+
+	appendFilesInDir := func(path string) error {
+		fis, err := ioutil.ReadDir(path)
+		if os.IsNotExist(err) {
+			return nil
+		}
+		if err != nil {
+			return errors.Wrapf(err, "failed to list files in dir '%s'", path)
+		}
+
+		for _, fi := range fis {
+			appendIfFile(filepath.Join(path, fi.Name()), fi)
+		}
+		return nil
+	}
+
+	if err := l.eachBuildpack(func(path string) error {
+		absPath, err := filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+		return eachDir(absPath, func(path string) error {
+			if err := appendFilesInDir(filepath.Join(path, dirName)); err != nil {
+				return err
+			}
+			if process.Type != "" {
+				return appendFilesInDir(filepath.Join(path, dirName, process.Type))
+			}
+			return nil
+		})
+	}, filters...); err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
+func (l *Launcher) eachBuildpack(fn func(path string) error, filters ...func(bp Buildpack) bool) error {
+Loop:
+	for _, bp := range l.Buildpacks {
+		for _, filter := range filters {
+			if !filter(bp) {
+				continue Loop
```

Comment:
Do you want to get back to line 215? Meaning, go to the next buildpack in `l.Buildpacks`?
If yes, isn't it better to call `break` here?